internal/rpc/jsonrpc/methods.go

```diff
@@ -2049,6 +2050,113 @@ func (s *Server) getTransaction(ctx context.Context, icmd interface{}) (interfac
 	return ret, nil
 }
 
+// getTxOut handles a gettxout request by returning details about an unspent
+// output. In SPV mode, details are only returned for transaction outputs that
+// are relevant to the wallet.
+// To match the behavior in RPC mode, (nil, nil) is returned if the transaction
+// output could not be found (never existed or was pruned) or is spent by another
+// transaction already in the main chain.  Mined transactions that are spent by
+// a mempool transaction are not affected by this.
+func (s *Server) getTxOut(ctx context.Context, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*types.GetTxOutCmd)
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, errUnloadedWallet
+	}
+
+	// Attempt RPC passthrough if connected to DCRD.
+	n, err := w.NetworkBackend()
+	if err != nil {
+		return nil, err
+	}
+	if rpc, ok := n.(*dcrd.RPC); ok {
+		var resp json.RawMessage
+		err := rpc.Call(ctx, "gettxout", &resp, cmd.Txid, cmd.Vout, cmd.Tree, cmd.IncludeMempool)
+		return resp, err
+	}
+
+	txHash, err := chainhash.NewHashFromStr(cmd.Txid)
+	if err != nil {
+		return nil, rpcError(dcrjson.ErrRPCDecodeHexString, err)
+	}
+
+	if cmd.Tree != wire.TxTreeRegular && cmd.Tree != wire.TxTreeStake {
+		return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter, "Tx tree must be regular or stake")
+	}
+
+	// Check if the transaction is known to wallet.
+	walletTx, err := wallet.UnstableAPI(w).TxDetails(ctx, txHash)
+	if err != nil && !errors.Is(err, errors.NotExist) {
+		return nil, err
+	}
+	if walletTx == nil {
+		return nil, nil // Tx not found in wallet.
+	}
+	if len(walletTx.MsgTx.TxOut) <= int(cmd.Vout) {
+		return nil, rpcErrorf(dcrjson.ErrRPCInvalidTxVout, "invalid vout %d", cmd.Vout)
+	}
+
+	tx := &walletTx.MsgTx
+	txTree := wire.TxTreeRegular
+	if stake.DetermineTxType(tx, true) != stake.TxTypeRegular {
+		txTree = wire.TxTreeStake
+	}
+	if txTree != cmd.Tree {
+		// Not an error because it is technically possible (though extremely unlikely)
+		// that the required tx (same hash, different tree) exists on the blockchain.
+		return nil, nil
+	}
+
+	// Attempt to read the unspent txout info from wallet.
+	outpoint := wire.OutPoint{Hash: *txHash, Index: cmd.Vout, Tree: cmd.Tree}
+	walletUnspent, err := w.UnspentOutput(ctx, outpoint, *cmd.IncludeMempool)
+	if err != nil && !errors.Is(err, errors.NotExist) {
+		return nil, err
+	}
+	if walletUnspent == nil {
```

Comment:
Just figured I'd point out that other than the nil check, `walletUnspent` is not used.  I can see `(*Wallet).UnspentOutput` being used elsewhere, so no issue really, just an observation in case you had more in mind for the full `udb.Credit`.  If not I wonder if `IsUnspentOutpoint` with modification to include/exclude mempool could be of use.

Reply:
`IsUnspentOutpoint` returns false if the output is spent in a mempool tx, which differs from the dcrd behaviour for gettxout. I've put `walletUnspent` to use instead to avoid modifying the existing `IsUnspentOutpoint` method.